File: blog/src/main/java/com/yky/blog/common/entity/OperationLog.java
package com.yky.blog.common.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("operation_log")
public class OperationLog {

    @TableId(type = IdType.AUTO)
    private Long id;
    private Long adminId;
    private String adminName;
    private String module;
    private String operation;
    private String requestMethod;
    private String requestPath;
    private String requestParams;
    private Integer responseCode;
    private String responseMessage;
    private String ip;
    private String userAgent;
    private Long costTime;
    private Integer success;
    private String errorMessage;
    private LocalDateTime createTime;
}

File: blog-admin/src/api/operationLog.ts
import request from '@/utils/request'

export interface OperationLog {
  id: number
  adminId: number | null
  adminName: string
  module: string
  operation: string
  requestMethod: string
  requestPath: string
  requestParams: string
  responseCode: number
  responseMessage: string
  ip: string
  userAgent: string
  costTime: number
  success: number
  errorMessage: string
  createTime: string
}

export interface OperationLogPageParams {
  page: number
  size: number
  module?: string
  operation?: string
  success?: number
  startTime?: string
  endTime?: string
}

export const getOperationLogPage = (params: OperationLogPageParams) =>
  request.get('/admin/operation-log/page', { params })
